BlueprintUI/Sources/Animation/Transition.swift

```diff
@@ -0,0 +1,84 @@
+import UIKit
+
+
+/**
+ Wraps a content element and adds transitions when the element appears, disappears, or changes layout.
+ */
+public struct Transition<Wrapped:Element> : Element {
+
+    /// The transition to use when the wrapped element is first presented on screen.
+    public var onAppear: TransitionAnimation?
+    
+    /// The transition to use when the wrapped element is removed from the screen.
+    public var onDisappear: TransitionAnimation?
+    
+    /// The layout transition to use when an already on-screen element is updated.
+    public var layoutTransition: LayoutTransition
+
+    /// The element which will be transitioned in, out, or transitioned during layout.
+    public var wrapped : Wrapped
+    
+    //
+    // MARK: Initialization
+    //
+    
+    /// Creates a new `Transition` with both `onAppear` and `onDisappear` set to the provided `animation`.
+    public init(
+        animation: TransitionAnimation,
+        layout: LayoutTransition = .specific(AnimationOptions()),
+        wrapping element: Wrapped
+    ) {
+        self.init(
+            onAppear: animation,
+            onDisappear: animation,
+            layout: layout,
+            wrapping: element
+        )
+    }
+
+    /**
+     Creates a new `Transition` with the provided options.
+    
+     If you only provide one of `onAppear` or `onDisappear`, only that direction of appearance
+     transition will be animated. The other one will be performed with no animation.
+     */
+    public init(
```

Comment:
We need to assert in here that the wrapped element is view backed (and maybe it’s type matches the animation?) for custom animations. Otherwise there could be multiple child views in the transition view.

Reply:
Or – likely smarter, wrap this in an element that _is_ view backed, which then drives animations for all children.

Reply:
- [ ] wrap in view backed element

Reply:
Wait, nvm.. this eventually will reduce down to a single view backed element.. I think? Need to verify this.